source/MaterialXGenShader/TypeDesc.h

```diff
@@ -57,6 +57,12 @@ class MX_GENSHADER_API TypeDesc
     static const TypeDesc* registerType(const string& name, unsigned char basetype, unsigned char semantic = SEMANTIC_NONE,
                                         size_t size = 1, bool editable = true, const ChannelMap& channelMapping = ChannelMap());
 
+    /// Equality operator overload
+    bool operator==(const TypeDesc& other) const;
```

Comment:
For consistency with other comparison operators in MaterialX, I would recommend naming this input `rhs`:

https://github.com/AcademySoftwareFoundation/MaterialX/blob/main/source/MaterialXCore/Types.h#L73

Reply:
Fixed.